src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -338,6 +351,29 @@ public GenotypeLikelihoodsAllelePair(final int alleleIndex1, final int alleleInd
         return cache;
     }
 
+
+    /**
+     * Calculate the cache of anyploid alleles for each PL index
+     * Creates the ordering for all possible combinations of ploidy alleles. Computed recursively and the
+     * result is stored in a cache.
+     *
+     * @param numAlleles    Number of alleles (including ref)
+     * @param ploidy        Number of chromosomes in set
+     * @param genotype      An entry of ploidy alleles
+     */
+    protected static void calculatePLcache(final int numAlleles, final int ploidy, final List<Integer> genotype){
+        if (genotype.size() == ploidy) {
+            plIndexToAlleleIndices.add(genotype);
+        }
+        else {
+            for (int a=0; a < numAlleles; a++) {
+                List gt = new ArrayList<Integer>(Arrays.asList(a));
+                gt.addAll(genotype);
```

Comment:
This will help: refer to "Algorithm for enumerating the genotypes given ploidy and alleles" in  http://genome.sph.umich.edu/wiki/Relationship_between_Ploidy,_Alleles_and_Genotypes